alodataset/kitti_odometry.py

```diff
@@ -0,0 +1,288 @@
+                raise ValueError(f"Sequence {sequences} does not exist")
+            if (int(sequences) <= 10 and self.split == Split.VAL) or (
+                int(sequences) >= 11 and self.split == Split.TRAIN
+            ):
+                raise ValueError(f"Sequence {sequences} is not in the {self.split} split")
+            loaded_sequences.append(sequences)
+        elif isinstance(sequences, list):
+            for seq in sequences:
+                if not os.path.exists(os.path.join(self.dataset_dir, "sequences", seq)):
+                    raise ValueError(f"Sequence {seq} does not exist")
+                if (int(seq) <= 10 and self.split == Split.VAL) or (int(seq) >= 11 and self.split == Split.TRAIN):
+                    raise ValueError(f"Sequence {seq} is not in the {self.split} split")
+                loaded_sequences.append(seq)
+
+        return loaded_sequences
+
+    def getitem(self, idx: int):
+        """
+        Loads a single frame from the dataset.
+
+        Parameters
+        ----------
+        idx : int
+            Index of the frame to load.
+
+        Returns
+        -------
+        Dict[str, Frame]
+            Dictionary with the loaded frame and pose.
+        """
+
+        if self.sample:
+            return BaseDataset.__getitem__(self, idx)
+
+        item = self.items[idx]
+
+        left = []
+        right = []
+        poses = []
+        sequence: str = item["sequence"]  # Number of the sequence
+
+        for id, seq in enumerate(item["temporal_sequence"]):
+
+            left.append(
+                Frame(
+                    os.path.join(
+                        self.dataset_dir,
+                        "sequences",
+                        sequence,
+                        f"image_{0 if self.grayscale else 2}",
+                        f"{seq:06d}.png",
+                    )
+                ).temporal()
+            )
+
+            if self.split == Split.TRAIN:
+                pose = Pose(
+                    torch.Tensor([float(x) for x in item["poses"][id].split(" ")] + [0, 0, 0, 1]).reshape(4, 4)
+                )
+                poses.append(pose)
+
+            if self.right_frame:
+                right.append(
+                    Frame(
+                        os.path.join(
+                            self.dataset_dir,
+                            "sequences",
+                            sequence,
+                            f"image_{1 if self.grayscale else 3}",
+                            f"{seq:06d}.png",
+                        )
+                    ).temporal()
+                )
+
+        frames = {}
+        frames["left"] = torch.cat(left, dim=0)
+        frames["left"].timestamp = item["times"]
+        frames["left"].baseline = self.seq_params["baseline"]
+        frames["left"].append_cam_intrinsic(self.seq_params["left_intrinsic"])
+        frames["left"].append_cam_extrinsic(self.seq_params["left_extrinsic"])
```

Comment:
See the previous comment. The calibration (baseline, intrinsic, extrinsic) are different for different sequences. You should use the `sequence` variable to read these values.